kyo-core/shared/src/main/scala/kyo/StreamCoreExtensions.scala

```diff
@@ -61,6 +62,40 @@ object StreamCoreExtensions:
                             _ <- emitMaybeChunksFromChannel(channel)
                         yield ()
 
+        /** Creates a stream from an iterator.
+          *
+          * @param v
+          *   Iterator to create a stream from
+          * @param bufferSize
+          *   Size of the buffer that the iterator will write to and the stream will read from
+          */
+        def fromIterator[V, S](v: => Iterator[V], bufferSize: Int = Stream.DefaultChunkSize)(using
+            tag: Tag[Emit[Chunk[V]]],
+            frame: Frame
+        ): Stream[V, IO] =
+            Stream:
+                IO:
+                    val it      = v
+                    val size    = bufferSize max 1
+                    val builder = ChunkBuilder.init[V]
+                    val nextElements: Chunk[V] < IO = IO:
+                        var count = 0
+                        while count < size && it.hasNext do
+                            builder.addOne(it.next())
+                            count += 1
+
+                        val res = builder.result()
+                        builder.clear()
+                        res
+
+                    Loop.foreach[Unit, IO & Emit[Chunk[V]]]:
+                        nextElements.map(chunk =>
+                            if chunk.isEmpty then Loop.done
+                            else Emit.valueWith(chunk)(Loop.continue)
+                        )
+
+        end fromIterator
```

Comment:
@hearnadam this is interesting, the previous implementation (using repeatPresent) had no issues with choice, this once has (`bufferSize = 1`):
```
[info]   fromIterator
[info]     bufferSize = 1
[info]     - call by name (1 millisecond)
[info]     - map with Choice *** FAILED *** (0 milliseconds)
[info]       Chunk.Indexed(Chunk.Indexed("A", "B", "C"), Chunk.Indexed("A", "B", "c"), Chunk.Indexed("A", "b"), Chunk.Indexed("a")) had size 4 instead of expected size 8 (StreamCoreExtensionsTest.scala:274)
[info]     bufferSize = 4
[info]     - call by name (4 milliseconds)
[info]     - map with Choice (4 milliseconds)
```

(however `repeatPresent` would fail the callByName test)